core/deployment/src/main/java/io/quarkus/deployment/steps/SubstrateConfigBuildStep.java

```diff
@@ -72,7 +74,7 @@ void build(SslContextConfigurationTemplate sslContextConfigurationTemplate,
         // it's probably overly conservative but it's a first step in the right direction
         sslContextConfigurationTemplate.setSslNativeEnabled(!sslNativeConfig.isExplicitlyDisabled());
 
-        if (sslNativeEnabled) {
+        if (sslNativeEnabled || !jniBuildItems.isEmpty()) {
```

Comment:
I don't think we should merge the two ifs. The logic is totally different.

Can you move the JNI stuff to a separate if?